R/create-population.R

```diff
@@ -3,15 +3,50 @@
 #' @param populationCharacteristics Characteristics of the population to create as an instance of \code{OriginData}
 #' that are actually distributed parameters
 #'
-#' @return An instance of a population object
+#' @return An list with two entries:
+#' The \code{population} An instance of a population object.
+#' The \code{derivedParameters} containing the parameter values modified indirectly by the algorithm. Those parameters are typically formula parameters.
 #'
 #' @export
 createPopulation <- function(populationCharacteristics) {
   validateIsOfType(populationCharacteristics, PopulationCharacteristics)
 
   populationFactory <- rClr::clrCallStatic("PKSim.R.Api", "GetPopulationFactory")
-  population <- rClr::clrCall(populationFactory, "CreatePopulation", populationCharacteristics$ref)
-  Population$new(population)
+  netPopulation <- rClr::clrCall(populationFactory, "CreatePopulation", populationCharacteristics$ref)
+  population <- Population$new(netPopulation)
+
+  individualCharacteristics <- NULL
+
+  #NOTE THIS IS A WORKAROUND UNTIL THE CODE IN PKSIM IS UPDATED
+  if (populationCharacteristics$species == Species$Human) {
+    # create an individual with similar properites Species and population. WEIGHT AND AGE DO NOT MATTER as long as we can create an invidiual
+    individualCharacteristics <- ospsuite::createIndividualCharacteristics(
+      species = populationCharacteristics$species,
+      population = populationCharacteristics$population,
+      weight = 70,
+      age = 30
+    )
+  }
+  else {
+    # create an individual with similar properites Species and population. WEIGHT AND AGE DO NOT MATTER as long as we can create an invidiual
+    individualCharacteristics <- ospsuite::createIndividualCharacteristics(
+      species = populationCharacteristics$species,
+      population = populationCharacteristics$population,
+      weight = populationCharacteristics$weightMin
+    )
+  }

```

Comment:
why do we need this differentiation between human and not-human here?
`individualCharacteristics `defined as in lines 24-27 would work also for animals, or not?

Reply:
maybe but I do not want ot create a 70kg mouse. I'd rather use the default from the DB.

Reply:
age is required for human however.

Reply:
The requirement for age has been added at some point - it was possible to create an individual without specifying the age. Why is it mandatory now?

Reply:
It's been like that in PKSim.Matlab for a while.  What is the use case of creating an individual without knowing the age?

Reply:
Creating a "standard" individual :) It did definitely work without specifying the age, I had it in one of our scripts and had to update it yesterday as I used it for the first time with the latest release of osps-R.